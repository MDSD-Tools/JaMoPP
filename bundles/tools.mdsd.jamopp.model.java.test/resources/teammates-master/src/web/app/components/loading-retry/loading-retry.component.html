<div *ngIf="shouldShowRetry; else content" class="text-center">
  <p class="text-muted m-1">{{message}}</p>
  <button (click)="retryEvent.emit()" class="btn btn-primary"><i class="fas fa-redo" aria-hidden="true"></i>&nbsp;Retry</button>
</div>

<ng-template #content>
  <ng-content></ng-content>
</ng-template>
