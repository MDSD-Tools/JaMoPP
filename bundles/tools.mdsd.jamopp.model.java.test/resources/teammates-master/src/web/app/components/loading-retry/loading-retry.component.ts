import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';

/**
 * Retry button for when loading fails.
 */
@Component({
  selector: 'tm-loading-retry',
  templateUrl: './loading-retry.component.html',
  styleUrls: ['./loading-retry.component.scss'],
})
export class LoadingRetryComponent implements OnInit {

  @Input() message: string = '';

  @Input() shouldShowRetry: boolean = false;

  @Output() retryEvent: EventEmitter<any> = new EventEmitter<any>();

  constructor() { }

  ngOnInit(): void {
  }

}
