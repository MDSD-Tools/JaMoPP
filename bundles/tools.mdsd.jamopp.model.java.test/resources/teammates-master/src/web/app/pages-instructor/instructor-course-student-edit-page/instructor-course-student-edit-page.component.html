<tm-loading-retry [shouldShowRetry]="hasStudentLoadingFailed" [message]="'Failed to load student'" (retryEvent)="loadStudentEditDetails(courseId, studentEmail)">
  <div *tmIsLoading="isStudentLoading">
    <div *ngIf="student" class="card border-primary">
      <div class="card-body fill-plain text-sm-left">
        <form
          id="instructor-student-edit-form"
          class="form form-horizontal"
          (ngSubmit)="onSubmit( resendPastLinksModal)" [formGroup]="editForm">
          <div class="form-row">
            <label class="col-sm-2">Course ID:</label>
            <div class="col-sm-10" id="courseid">{{ courseId }}</div>
          </div>
          <div class="form-row">
            <label class="col-sm-2 col-form-label">Student Name:</label>
            <div class="col-sm-10">
              <input
                class="form-control"
                id="studentname"
                formControlName="studentname">
                <div
                  *ngIf="editForm.get('studentname')!.errors
                  && (editForm.get('studentname')!.dirty || editForm.get('studentname')!.touched)"
                  class="invalid-field">
                  <i class="fa fa-exclamation-circle" aria-hidden="true"></i>
                  <span [hidden]="!editForm.get('studentname')!.errors!.required">
                    {{ displayEmptyFieldMessage('Student Name') }}
                  </span>
                  <span [hidden]="!editForm.get('studentname')!.errors!.maxlength">
                    {{ displayExceedMaxLengthMessage('Student Name', FormValidator.STUDENT_NAME_MAX_LENGTH) }}
                  </span>
                </div>
            </div>
          </div>
          <div class="form-row">
            <label class="col-sm-2 col-form-label">Section Name:</label>
            <div class="col-sm-10">
              <input
                class="form-control"
                id="sectionname"
                formControlName="sectionname">
              <div
                *ngIf="editForm.get('sectionname')!.errors
                && (editForm.get('sectionname')!.dirty || editForm.get('sectionname')!.touched)"
                class="invalid-field">
                <i class="fa fa-exclamation-circle" aria-hidden="true"></i>
                <span [hidden]="!editForm.get('sectionname')!.errors!.required">
                  {{ displayEmptyFieldMessage('Section Name') }}
                </span>
                <span [hidden]="!editForm.get('sectionname')!.errors!.maxlength">
                  {{ displayExceedMaxLengthMessage('Section Name', FormValidator.SECTION_NAME_MAX_LENGTH) }}
                </span>
              </div>
            </div>
          </div>
          <div class="form-row">
            <label class="col-sm-2 col-form-label">Team Name:</label>
            <div class="col-sm-10">
              <input
                class="form-control col-form-label"
                name="teamname"
                id="teamname"
                formControlName="teamname">
              <div
                *ngIf="editForm.get('teamname')!.errors
                && (editForm.get('teamname')!.dirty || editForm.get('teamname')!.touched)"
                class="invalid-field">
                <i class="fa fa-exclamation-circle" aria-hidden="true"></i>
                <span [hidden]="!editForm.get('teamname')!.errors!.required">
                  {{ displayEmptyFieldMessage('Team Name') }}
                </span>
                <span [hidden]="!editForm.get('teamname')!.errors!.maxlength">
                  {{ displayExceedMaxLengthMessage('Team Name', FormValidator.TEAM_NAME_MAX_LENGTH) }}
                </span>
              </div>
            </div>
          </div>
          <div class="form-row">
            <label class="col-sm-2 col-form-label">E-mail Address:</label>
            <div class="col-sm-10">
              <input
                class="form-control"
                id="newstudentemail"
                formControlName="newstudentemail">
              <div
                *ngIf="editForm.get('newstudentemail')!.errors
                && (editForm.get('newstudentemail')!.dirty || editForm.get('newstudentemail')!.touched)"
                  class="invalid-field">
                <i class="fa fa-exclamation-circle" aria-hidden="true"></i>
                <span [hidden]="!editForm.get('newstudentemail')!.errors!.required">
                  {{ displayEmptyFieldMessage('E-mail Address') }}
                </span>
                <span [hidden]="!editForm.get('newstudentemail')!.errors!.maxlength">
                  {{ displayExceedMaxLengthMessage('E-mail Address', FormValidator.EMAIL_MAX_LENGTH) }}
                </span>
              </div>
            </div>
          </div>
          <div class="form-row">
            <label class="col-sm-2 col-form-label">Comments:</label>
            <div class="col-sm-10">
              <textarea
                class="form-control"
                rows="6"
                id="comments"
                formControlName="comments">
              </textarea>
            </div>
          </div>
          <br>
          <div class="text-center">
            <button class="btn btn-success"
                    id="btn-submit"
                    type="submit"
                    [disabled]="!editForm.valid || isFormSaving">
              <i [ngClass]="isFormSaving ? 'spinner-border spinner-border-sm' : 'fas fa-save'"></i>
              Save Changes
            </button>
          </div>
          <br>
          <br>

          <ng-template #resendPastLinksModal let-modal>
            <div class="modal-header alert-primary">
              <h4 class="modal-title icon-primary">Resend past links to the new email?</h4>
              <button
                  type="button"
                  class="close"
                  data-dismiss="modal"
                  aria-hidden="true"
                  (click)="modal.dismiss()">Ã—</button>
            </div>
            <div class="modal-body">
              Do you want to resend past session links of this course to the new email {{ editForm.get('newstudentemail')?.value }}?
            </div>
            <div class="modal-footer">
              <button id="btn-resend-links" type="button" class="btn btn-primary"
                      (click)="submitEditForm(true); modal.dismiss()">
                Yes, save changes and resend links
              </button>
              <button type="button" class="btn btn-primary"
                      (click)="submitEditForm(false); modal.dismiss()">
                No, just save the changes
              </button>
              <button type="button" class="btn btn-outline-secondary"
                      (click)="modal.dismiss()">
                Cancel
              </button>
            </div>
          </ng-template>
        </form>
      </div>
    </div>
  </div>
</tm-loading-retry>
