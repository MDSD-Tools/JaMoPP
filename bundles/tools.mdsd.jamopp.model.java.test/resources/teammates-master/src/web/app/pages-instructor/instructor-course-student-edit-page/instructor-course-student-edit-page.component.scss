.fill-plain {
  background-color: #EAEFF5;
}

label {
  font-weight: bold;
}

.form-row {
  margin-bottom: 15px;
}

.invalid-field {
  padding-top: 5px;
  color: #B50000;
}
