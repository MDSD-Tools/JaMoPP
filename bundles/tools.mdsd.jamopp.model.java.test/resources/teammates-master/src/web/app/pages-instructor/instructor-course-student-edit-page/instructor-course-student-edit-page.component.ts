import { Component, Input, OnDestroy, OnInit } from '@angular/core';
import { AbstractControl, FormControl, FormGroup, Validators } from '@angular/forms';
import { ActivatedRoute, Router } from '@angular/router';
import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
import { Subscription } from 'rxjs';
import { finalize } from 'rxjs/operators';
import { StatusMessageService } from '../../../services/status-message.service';
import { JoinState, MessageOutput, Student } from '../../../types/api-output';
import { StudentUpdateRequest } from '../../../types/api-request';
import { ErrorMessageOutput } from '../../error-message-output';

import { NavigationService } from '../../../services/navigation.service';
import { SimpleModalService } from '../../../services/simple-modal.service';
import { StudentService } from '../../../services/student.service';
import { FormValidator } from '../../../types/form-validator';
import { SimpleModalType } from '../../components/simple-modal/simple-modal-type';

/**
 * Instructor course student edit page.
 */
@Component({
  selector: 'tm-instructor-course-student-edit-page',
  templateUrl: './instructor-course-student-edit-page.component.html',
  styleUrls: ['./instructor-course-student-edit-page.component.scss'],
})
export class InstructorCourseStudentEditPageComponent implements OnInit, OnDestroy {

  FormValidator: typeof FormValidator = FormValidator; // enum

  @Input() isEnabled: boolean = true;
  courseId: string = '';
  studentEmail: string = '';
  student!: Student;

  isTeamnameFieldChanged: boolean = false;
  isEmailFieldChanged: boolean = false;
  isStudentLoading: boolean = false;
  hasStudentLoadingFailed: boolean = false;
  isFormSaving: boolean = false;

  editForm!: FormGroup;
  teamFieldSubscription?: Subscription;
  emailFieldSubscription?: Subscription;

  constructor(private route: ActivatedRoute,
              private router: Router,
              private statusMessageService: StatusMessageService,
              private studentService: StudentService,
              private navigationService: NavigationService,
              private ngbModal: NgbModal,
              private simpleModalService: SimpleModalService) { }

  ngOnInit(): void {
    if (!this.isEnabled) {
      this.student = {
        email: 'alice@email.com',
        courseId: '',
        name: 'Alice Betsy',
        comments: 'Alice is a transfer student.',
        teamName: 'Team A',
        sectionName: 'Section A',
        joinState: JoinState.JOINED,
      };
      this.initEditForm();
      return;
    }

    this.route.queryParams.subscribe((queryParams: any) => {
      this.courseId = queryParams.courseid;
      this.studentEmail = queryParams.studentemail;
      this.loadStudentEditDetails(queryParams.courseid, queryParams.studentemail);
    });
  }

  ngOnDestroy(): void {
    if (this.emailFieldSubscription) {
      (this.emailFieldSubscription as Subscription).unsubscribe();
    }
    if (this.teamFieldSubscription) {
      (this.teamFieldSubscription as Subscription).unsubscribe();
    }
  }

  /**
   * Loads student details required for this page.
   */
  loadStudentEditDetails(courseId: string, studentEmail: string): void {
    this.hasStudentLoadingFailed = false;
    this.isStudentLoading = true;
    this.studentService.getStudent(
        courseId, studentEmail,
    ).pipe(finalize(() => this.isStudentLoading = false)).subscribe((student: Student) => {
      this.student = student;
      this.initEditForm();
    }, (resp: ErrorMessageOutput) => {
      this.hasStudentLoadingFailed = true;
      this.statusMessageService.showErrorToast(resp.error.message);
    });
  }

  /**
   * Initializes the student details edit form with the fields fetched from the backend.
   * Subscriptions are set up to listen to changes in the 'teamname' fields and 'newstudentemail' fields.
   */
  private initEditForm(): void {
    this.editForm = new FormGroup({
      studentname: new FormControl(this.student.name,
          [Validators.required, Validators.maxLength(FormValidator.STUDENT_NAME_MAX_LENGTH)]),
      sectionname: new FormControl(this.student.sectionName,
          [Validators.required, Validators.maxLength(FormValidator.SECTION_NAME_MAX_LENGTH)]),
      teamname: new FormControl(this.student.teamName,
          [Validators.required, Validators.maxLength(FormValidator.TEAM_NAME_MAX_LENGTH)]),
      newstudentemail: new FormControl(this.student.email, // original student email initialized
          [Validators.required, Validators.maxLength(FormValidator.EMAIL_MAX_LENGTH)]),
      comments: new FormControl(this.student.comments),
    });
    this.teamFieldSubscription =
        (this.editForm.get('teamname') as AbstractControl).valueChanges
            .subscribe(() => {
              this.isTeamnameFieldChanged = true;
            });

    this.emailFieldSubscription =
        (this.editForm.get('newstudentemail') as AbstractControl).valueChanges
            .subscribe(() => this.isEmailFieldChanged = true);
  }

  /**
   * Displays message to user stating that the field is empty.
   */
  displayEmptyFieldMessage(fieldName: string): string {
    return `The field '${fieldName}' should not be empty.`;
  }

  /**
   * Displays message to user stating that the field exceeds the max length.
   */
  displayExceedMaxLengthMessage(fieldName: string, maxLength: number): string {
    return `The field '${fieldName}' should not exceed ${maxLength} characters.`;
  }

  /**
   * Handles logic related to showing the appropriate modal boxes
   * upon submission of the form. Submits the form otherwise.
   */
  onSubmit(resendPastLinksModal: any): void {
    if (!this.isEnabled) {
      return;
    }

    if (this.isTeamnameFieldChanged) {
      const modalContent: string = `Editing these fields will result in some existing responses from this student to be deleted.
            You may download the data before you make the changes.`;
      const modalRef: NgbModalRef = this.simpleModalService.openConfirmationModal(
          'Delete existing responses?', SimpleModalType.WARNING, modalContent);
      modalRef.result.then(() => {
        this.deleteExistingResponses(resendPastLinksModal);
      }, () => {});
    } else if (this.isEmailFieldChanged) {
      this.ngbModal.open(resendPastLinksModal);
    } else {
      this.submitEditForm(false);
    }
  }

  /**
   * Shows the `resendPastSessionLinks` modal if email field has changed.
   * Submits the form  otherwise.
   */
  deleteExistingResponses(resendPastLinksModal: any): void {
    if (this.isEmailFieldChanged) {
      this.ngbModal.open(resendPastLinksModal);
    } else {
      this.submitEditForm(false);
    }
  }

  /**
   * Submits the form data to edit the student details.
   */
  submitEditForm(shouldResendPastSessionLinks: boolean): void {
    const reqBody: StudentUpdateRequest = {
      name: this.editForm.value.studentname,
      email: this.editForm.value.newstudentemail,
      team: this.editForm.value.teamname,
      section: this.editForm.value.sectionname,
      comments: this.editForm.value.comments,
      isSessionSummarySendEmail: shouldResendPastSessionLinks,
    };

    this.isFormSaving = true;

    this.studentService.updateStudent({
      courseId: this.courseId,
      studentEmail: this.student.email,
      requestBody: reqBody,
    })
      .pipe(finalize(() => {
        this.isFormSaving = false;
      }))
      .subscribe((resp: MessageOutput) => {
        this.navigationService.navigateWithSuccessMessage(this.router, '/web/instructor/courses/details',
            resp.message, { courseid: this.courseId });
      }, (resp: ErrorMessageOutput) => {
        this.statusMessageService.showErrorToast(resp.error.message);
      });
  }
}
