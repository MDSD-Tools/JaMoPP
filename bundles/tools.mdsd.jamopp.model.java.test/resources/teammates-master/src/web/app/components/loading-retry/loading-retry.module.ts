import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { LoadingRetryComponent } from './loading-retry.component';

/**
 * Module for the retry button
 */
@NgModule({
  declarations: [LoadingRetryComponent],
  imports: [
    CommonModule,
  ],
  exports: [LoadingRetryComponent],
})
export class LoadingRetryModule { }
