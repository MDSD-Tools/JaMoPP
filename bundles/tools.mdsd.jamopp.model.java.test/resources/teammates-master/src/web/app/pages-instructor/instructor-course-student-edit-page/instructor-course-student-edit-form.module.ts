import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { InstructorCourseStudentEditPageComponent } from './instructor-course-student-edit-page.component';

/**
 * Module for course student edit form.
 */
@NgModule({
  declarations: [
    InstructorCourseStudentEditPageComponent,
  ],
  exports: [
    InstructorCourseStudentEditPageComponent,
  ],
  imports: [
    CommonModule,
    ReactiveFormsModule,
    LoadingSpinnerModule,
    LoadingRetryModule,
  ],
})
export class InstructorCourseStudentEditFormModule { }
