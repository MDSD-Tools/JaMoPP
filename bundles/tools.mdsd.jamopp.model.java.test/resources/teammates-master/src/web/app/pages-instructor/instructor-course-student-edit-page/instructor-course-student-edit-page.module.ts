import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { InstructorCourseStudentEditFormModule } from './instructor-course-student-edit-form.module';
import { InstructorCourseStudentEditPageComponent } from './instructor-course-student-edit-page.component';

const routes: Routes = [
  {
    path: '',
    component: InstructorCourseStudentEditPageComponent,
  },
];

/**
 * Module for instructor course student edit page.
 */
@NgModule({
  imports: [
    CommonModule,
    InstructorCourseStudentEditFormModule,
    RouterModule.forChild(routes),
  ],
})
export class InstructorCourseStudentEditPageModule { }
