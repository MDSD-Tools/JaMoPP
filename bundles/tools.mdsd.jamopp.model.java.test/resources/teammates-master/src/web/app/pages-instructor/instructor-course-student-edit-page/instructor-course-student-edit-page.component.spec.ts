import { HttpClientTestingModule } from '@angular/common/http/testing';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
import { RouterTestingModule } from '@angular/router/testing';
import { JoinState } from '../../../types/api-output';
import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { InstructorCourseStudentEditPageComponent } from './instructor-course-student-edit-page.component';

describe('InstructorCourseStudentEditPageComponent', () => {
  let component: InstructorCourseStudentEditPageComponent;
  let fixture: ComponentFixture<InstructorCourseStudentEditPageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [InstructorCourseStudentEditPageComponent],
      imports: [
        RouterTestingModule,
        ReactiveFormsModule,
        HttpClientTestingModule,
        LoadingSpinnerModule,
        LoadingRetryModule,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InstructorCourseStudentEditPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with student details', () => {
    component.student = {
      email: 'jake@gmail.com',
      courseId: 'Crime101',
      name: 'Jake Peralta',
      comments: 'Cool cool cool.',
      teamName: 'Team A',
      sectionName: 'Section A',
      joinState: JoinState.JOINED,
    };
    component.editForm = new FormGroup({
      studentname: new FormControl('Jake Peralta'),
      sectionname: new FormControl('Section A'),
      teamname: new FormControl('Team A'),
      newstudentemail: new FormControl('jake@gmail.com'),
      comments: new FormControl('Cool cool cool.'),
    });
    component.isStudentLoading = false;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap when student is still loading', () => {
    component.isStudentLoading = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
