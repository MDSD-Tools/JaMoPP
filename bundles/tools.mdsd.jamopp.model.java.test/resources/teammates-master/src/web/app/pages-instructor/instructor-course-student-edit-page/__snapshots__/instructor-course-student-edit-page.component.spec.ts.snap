// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructorCourseStudentEditPageComponent should snap when student is still loading 1`] = `
<tm-instructor-course-student-edit-page
  FormValidator={[Function Object]}
  courseId="undefined"
  hasStudentLoadingFailed="false"
  isEmailFieldChanged="false"
  isEnabled={[Function Boolean]}
  isFormSaving="false"
  isStudentLoading={[Function Boolean]}
  isTeamnameFieldChanged="false"
  navigationService={[Function NavigationService]}
  ngbModal={[Function NgbModal]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  simpleModalService={[Function SimpleModalService]}
  statusMessageService={[Function StatusMessageService]}
  studentEmail="undefined"
  studentService={[Function StudentService]}
>
  <tm-loading-retry>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    
    
  </tm-loading-retry>
</tm-instructor-course-student-edit-page>
`;

exports[`InstructorCourseStudentEditPageComponent should snap with default fields 1`] = `
<tm-instructor-course-student-edit-page
  FormValidator={[Function Object]}
  courseId="undefined"
  hasStudentLoadingFailed="false"
  isEmailFieldChanged="false"
  isEnabled={[Function Boolean]}
  isFormSaving="false"
  isStudentLoading={[Function Boolean]}
  isTeamnameFieldChanged="false"
  navigationService={[Function NavigationService]}
  ngbModal={[Function NgbModal]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  simpleModalService={[Function SimpleModalService]}
  statusMessageService={[Function StatusMessageService]}
  studentEmail="undefined"
  studentService={[Function StudentService]}
>
  <tm-loading-retry>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    
    
  </tm-loading-retry>
</tm-instructor-course-student-edit-page>
`;

exports[`InstructorCourseStudentEditPageComponent should snap with student details 1`] = `
<tm-instructor-course-student-edit-page
  FormValidator={[Function Object]}
  courseId="undefined"
  editForm={[Function FormGroup]}
  hasStudentLoadingFailed="false"
  isEmailFieldChanged="false"
  isEnabled={[Function Boolean]}
  isFormSaving="false"
  isStudentLoading="false"
  isTeamnameFieldChanged="false"
  navigationService={[Function NavigationService]}
  ngbModal={[Function NgbModal]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  simpleModalService={[Function SimpleModalService]}
  statusMessageService={[Function StatusMessageService]}
  student={[Function Object]}
  studentEmail="undefined"
  studentService={[Function StudentService]}
>
  <tm-loading-retry>
    
    
    <div>
      
      <div
        class="card border-primary"
      >
        <div
          class="card-body fill-plain text-sm-left"
        >
          <form
            class="form form-horizontal ng-untouched ng-pristine ng-valid"
            id="instructor-student-edit-form"
            novalidate=""
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-2"
              >
                Course ID:
              </label>
              <div
                class="col-sm-10"
                id="courseid"
              >
                
              </div>
            </div>
            <div
              class="form-row"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Student Name:
              </label>
              <div
                class="col-sm-10"
              >
                <input
                  class="form-control ng-untouched ng-pristine ng-valid"
                  formcontrolname="studentname"
                  id="studentname"
                />
                
              </div>
            </div>
            <div
              class="form-row"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Section Name:
              </label>
              <div
                class="col-sm-10"
              >
                <input
                  class="form-control ng-untouched ng-pristine ng-valid"
                  formcontrolname="sectionname"
                  id="sectionname"
                />
                
              </div>
            </div>
            <div
              class="form-row"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Team Name:
              </label>
              <div
                class="col-sm-10"
              >
                <input
                  class="form-control col-form-label ng-untouched ng-pristine ng-valid"
                  formcontrolname="teamname"
                  id="teamname"
                  name="teamname"
                />
                
              </div>
            </div>
            <div
              class="form-row"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                E-mail Address:
              </label>
              <div
                class="col-sm-10"
              >
                <input
                  class="form-control ng-untouched ng-pristine ng-valid"
                  formcontrolname="newstudentemail"
                  id="newstudentemail"
                />
                
              </div>
            </div>
            <div
              class="form-row"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Comments:
              </label>
              <div
                class="col-sm-10"
              >
                <textarea
                  class="form-control ng-untouched ng-pristine ng-valid"
                  formcontrolname="comments"
                  id="comments"
                  rows="6"
                >
                                
                </textarea>
              </div>
            </div>
            <br />
            <div
              class="text-center"
            >
              <button
                class="btn btn-success"
                id="btn-submit"
                type="submit"
              >
                <i
                  class="fas fa-save"
                />
                 Save Changes 
              </button>
            </div>
            <br />
            <br />
            
          </form>
        </div>
      </div>
    </div>
    
    
  </tm-loading-retry>
</tm-instructor-course-student-edit-page>
`;
